packages/map-template/src/components/Map/Map.jsx

```diff
@@ -18,40 +34,37 @@ let _tileStyle;
  * Shows a map.
  *
  * @param {Object} props
- * @param {string} [props.apiKey] - MapsIndoors API key or solution alias.
- * @param {string} [props.gmApiKey] - Google Maps API key if you want to show a Google Maps map.
- * @param {string} [props.mapboxAccessToken] - Mapbox Access Token if you want to show a Mapbox map.
- * @param {array} [props.venues] - Array of Venues in the current solution.
- * @param {string} [props.venueName] - If you want the map to show a specific Venue, provide the Venue name here.
  * @param {function} [props.onLocationClick] - Function that is run when a MapsIndoors Location is clicked. the Location will be sent along as first argument.
- * @param {function} props.onMapsIndoorsInstance - Function that is run when a MapsIndoors instance is created. The instance will be sent along as first argument.
- * @param {function} props.onDirectionsService - Function that is run when a DirectionsService instance is created. The instance will be sent along as first argument.
  * @param {function} props.onVenueChangedOnMap - Function that is run when the map bounds was changed due to fitting to a venue.
- * @param {function} props.onPositionControl -  A function that is called when the MapsIndoors PositionControl is constructed. Will send the PositionControl instance as payload.
- * @param {function} props.onUserPosition - Function that is run when (if) the user position updates. Sends position as payload.
- * @param {array} props.filteredLocationIds - Array of IDs of the filtered locations.
- * @param {function} props.onMapTypeChanged - Function that is run when the map type is changed.
- * @param {array} props.filteredLocationsByExternalIDs - Array of IDs of the filtered locations based on external ID.
- * @param {string} props.tileStyle - Tile style name to change the interface of the map.
- * @param {number} props.startZoomLevel - The initial zoom level of the map.
- * @param {string} props.locationId - Location Id property used to handle the centering and zooming of the map.
-
  * @returns
  */
-function Map({ apiKey, gmApiKey, mapboxAccessToken, venues, venueName, onLocationClick, onMapsIndoorsInstance, onDirectionsService, onVenueChangedOnMap, onPositionControl, onUserPosition, filteredLocationIds, onMapTypeChanged, filteredLocationsByExternalIDs, tileStyle, startZoomLevel, locationId }) {
-    const [mapType, setMapType] = useState();
-    const [mapsIndoorsInstance, setMapsIndoorsInstance] = useState(null);
+function Map({ onLocationClick, onVenueChangedOnMap }) {
+    const apiKey = useRecoilValue(apiKeyState);
+    const gmApiKey = useRecoilValue(gmApiKeyState);
+    const mapboxAccessToken = useRecoilValue(mapboxAccessTokenState);
+    const [mapType, setMapType] = useRecoilState(mapTypeState);
+    const [mapsIndoorsInstance, setMapsIndoorsInstance] = useRecoilState(mapsIndoorsInstanceState);
+    const [, setUserPosition] = useRecoilState(userPositionState);
```

Comment:
Is this the pattern that needs to be followed when using the state, leaving the value empty?

Reply:
It's one pattern, unsure if is is "the" pattern 😄

Reply:
Would it still work if we would have a value like `userPosition` ?